### DISCUSSION TITLE
floorplanner.com - svg file

### DISCUSSION THREAD

**Discussion by theagilepadawan** [ExperienceLovelace/ha-floorplan#131] [Q&A] [answered]
I've created a floorplan on floorplanner.com and upped the project so that I'm able to get HD and PDF-exports. 

The approach I've tried is to convert the PDF I generate into a svg file. But inspecting the svg file further I see that the resulting svg is just embedding a png file. This not being a "proper" svg file with each room having a svg path and ID I'm not able to use the same approach as the guide outlines (E.G styling and animation per room and sensor)

Am I missing something here? Do I need an even better export at floorplanner.com?

All best,
Michael

**Comment by exetico** [comment, marked as answer]
You sadly can't export the room-layout and so. The floorplanner-image will therefore act as a background. You'll draw the boxes on top of that, for each room.

Check the Floorplanner SVG example-file from the docs, if you're still unsure about how to take the next step 👍.

**Reply by theagilepadawan** [reply]
Thanks, @exetico! 

I realised this the moment I posted the issue 🙈  Nevertheless, thanks for the swift reply!

**Comment by Ulrar** [comment]
Hi,

I'm confused, do you mean [this file](https://github.com/ExperienceLovelace/ha-floorplan/blob/master/docs/_docs/floorplan/examples/floorplanner_home/floorplanner_home.svg) ?
I went with floorplanner since the doc mentions it, assuming it had an export to svg function. But since it doesn't I don't really know what to do next.

Thanks

**Reply by OzGav** [reply]
Not sure exactly what you are having difficulty with? You need an SVG file that has an image of your house. Then on top of that you need to add SVG elements that you will use to “click” on to interact with or to dynamically change to show information.
How you generate the SVG is up to you. You can draw it from scratch (personally I use Sweet Home 3D) or you can just import an image and use it as a background.

**Reply by exetico** [reply]
Hi Ulrar

_The follwoing part are written in a Hurry. Hopefully it make sense._

As pr. my understanding, you're facing the same problem as theagilepadawan?

You'll have to create the "good"-looking floorplan **image** at Floorplanner.com. Hereafter, export it as a png-image (other formats works, too). Floorplanner sadly [doesn't support](https://twitter.com/floorplanner/status/78836211095044096) exports to SVG. So you'll have to strik to the Floorplanner work, as a background for your **own** SVG-file created in Inkscape. 

_Floorplan only supports these formats_:
![image](https://user-images.githubusercontent.com/3549445/142062449-ba05635e-9d88-426f-835b-3d14c76e3416.png)

After you've exported the floorplan image at Floorplanner.com, you'll have to fire up [Inkscape](https://experiencelovelace.github.io/ha-floorplan/docs/create-svg-file/), to create your first floorplan as a SVG-project.

See the Floorplanner.com image as the best option to make a beautify floorplan, and Inkscape to wrap/draw each room on top, as the "upper technical layer". 

If that's too complicated, no-one are preventing you from just using Inkscape, without a beautify background made at Floorplanner.com. Boxes placed next to each other will work fine. Check our basic home-example, if that's the right path for your.

Here's **my areas** in my SVG floorplan, created by manually adding elements in Inkscape. 
![image](https://user-images.githubusercontent.com/3549445/142065147-a99cac30-8a56-4e5b-b85f-69cbd1eb8630.png)
In the following view mode:
![image](https://user-images.githubusercontent.com/3549445/142065319-c48e0dc3-efe0-4383-ae2f-6278e96ab472.png)
These boxes are drawn manually.

It's best to use layers, simply because it's easier for your to navigate in the floorplan, if you're planning to add multiple rooms, things, texts and so on. 

The layers-sidebare can be triggered here:
![image](https://user-images.githubusercontent.com/3549445/142065471-cc3fe14c-c70b-4059-b2e4-42c3df363265.png)

Here's my mockup, without the Floorplanner.com background image and sensors:
![image](https://user-images.githubusercontent.com/3549445/142065571-eba23c11-161b-4a98-8b1b-4318c522839f.png)

Use the Objects-panel, to see each of my areas. 
![image](https://user-images.githubusercontent.com/3549445/142065674-6a811fa8-df14-4341-a3f8-76b8664b4647.png)
![image](https://user-images.githubusercontent.com/3549445/142065696-a17db225-8421-4f04-8d81-7a521f3519a7.png)

And oh! Object properties are added here. That's used to give the element a ID, which you'll use in the YAML-config later on. 
![image](https://user-images.githubusercontent.com/3549445/142065836-1d3256d7-6c6a-4863-b230-f42c22e5bc4b.png)

I simply added my Floorplanner-image are added as a image in Inkscape... :-) 

_I hope this will help you through - maybe just a bit. I'll consider to create a full tutorial at some point. But currently I simply don't have time. Therefore, I just grapped a few screenshots. Remember that my person SVG are in our repo - so you're also able to play around with that._

**Reply by Ulrar** [reply]
Hi @exetico, thanks for the write-up, that's exactly what I was looking for, how to get from floorplanner to an usable svg.
I'll give it a go tomorrow, thanks !

**Reply by exetico** [reply]
You're welcome! I've added a few more things, and corrected a few, too.

Good luck. The docs should help you just fine, with the config-things 👍.

**Reply by TheCherry** [reply]
The exploration from exetico, should take place into the documentation! Exactly something like this have to be there.

**Reply by OzGav** [reply]
You could cut and paste it to here @exetico ??
https://experiencelovelace.github.io/ha-floorplan/docs/create-svg-file/

**Reply by dal55555moo** [reply]
@exetico Have you put together a full tutorial by any chance? I've never used inkscape before so I'm not clear on how you set the .png as your background image? And then once it's your background image what are you supposed to do? Thanks for the work you've put into this but I think I'm missing a few steps?

**Reply by exetico** [reply]
I've sadly no 1 to 1 step guide at the moment, but I'm still considering creating a video-series from A to Z. But its hard to find the proper amount of time. With that said, i found myself Googling a bit if I run into problems with Inkscape. I only have a few hours in the problem. I'm more a coding-guy, so I dont have to get into yet another GUI 😊.

I'll suggest to grap my SVG file from our repo, and find the image object. I guess that will help you out? Its just a image moved behind all other elements, nothing fancy.

https://github.com/ExperienceLovelace/ha-floorplan/raw/master/docs/_docs/floorplan/examples/floorplanner_home/floorplanner_home.svg 

I really enjoy seeing all the advanced floorplans build with ha-floorplan, but I'm having a internal fight with the 24-hours, spend on multiple things all at once 😁.

**Reply by dal55555moo** [reply]
No worries, I know what it's like to have too many irons in the fire! I found a video that I'm hoping will walk me through things a bit :)

**Reply by SoloAsh** [reply]
I really appreciate this. I was not understanding that I needed to trace over the floorplan image with inkscape. Thanks!
